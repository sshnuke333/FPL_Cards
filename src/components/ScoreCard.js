import React from 'react';
import {
    ScoreDiv,
    ScoreCardDiv,
    ScoreCardLabel,
    ScoreCardDetails,
} from './ScoreCard.styles';

export const ScoreCard = ({ playerLeft, opponentLeft }) => {
    return (
        <ScoreCardDiv>
            <ScoreCardDetails>Cards Left</ScoreCardDetails>
            <ScoreCardLabel column="1/2">YOU</ScoreCardLabel>
            <ScoreDiv>
                <span>{playerLeft}</span>
                <span>-</span>
                <span>{opponentLeft}</span>
            </ScoreDiv>
            <ScoreCardLabel column="3/4">COMP</ScoreCardLabel>
        </ScoreCardDiv>
    );
};
