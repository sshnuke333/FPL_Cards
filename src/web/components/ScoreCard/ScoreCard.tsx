import {
    ScoreDiv,
    ScoreCardDiv,
    ScoreCardLabel,
    ScoreCardDetails,
} from './ScoreCard.styles';

interface IScoreCard {
    playerLeft: number;
    opponentLeft: number;
}
const ScoreCard = ({ playerLeft, opponentLeft }: IScoreCard): JSX.Element => (
    <ScoreCardDiv data-testid="scorecard">
        <ScoreCardDetails>Cards Left</ScoreCardDetails>
        <ScoreCardLabel column="1/2">YOU</ScoreCardLabel>
        <ScoreDiv>
            <span>{playerLeft}</span>
            <span>-</span>
            <span>{opponentLeft}</span>
        </ScoreDiv>
        <ScoreCardLabel column="3/4">COMP</ScoreCardLabel>
    </ScoreCardDiv>
);

export default ScoreCard;
