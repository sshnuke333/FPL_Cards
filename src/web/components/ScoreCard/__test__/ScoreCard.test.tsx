import { render } from '@testing-library/react';
import ScoreCard from '../ScoreCard';

describe('ScoreCard', () => {
    it('renders successfully', () => {
        const { asFragment } = render(
            <ScoreCard playerLeft="1" opponentLeft="1" />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
