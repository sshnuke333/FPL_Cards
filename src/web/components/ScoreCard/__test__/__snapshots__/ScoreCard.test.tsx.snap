// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreCard renders successfully 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe ktPVLR"
    data-testid="scorecard"
  >
    <div
      class="sc-gswNZR kQLhdT"
    >
      Cards Left
    </div>
    <div
      class="sc-dkrFOg iSUdvU"
    >
      YOU
    </div>
    <div
      class="sc-hLBbgP fgYGba"
    >
      <span>
        1
      </span>
      <span>
        -
      </span>
      <span>
        1
      </span>
    </div>
    <div
      class="sc-dkrFOg kwTiUo"
    >
      COMP
    </div>
  </div>
</DocumentFragment>
`;
